mask/eolearn/mask/snow_mask.py

```diff
@@ -0,0 +1,213 @@
+    Otherwise, if `data_feature` does not exist, a new OGC request at the given resolution is made and the
+    classifier is run. This design should allow faster
+    execution of the classifier, and reduce the number of requests. `linear` interpolation is used for resampling of
+    the `data_feature` and cloud probability map, while `nearest` interpolation is used to upsample the binary cloud
+    mask.
+    """
+    def __init__(self, data_feature, band_indices, cloud_mask_feature, dem_feature, dilation_size=0, r_f=12, d_z=100,
+                 r_b=0.1, r_d=0.3, n_1=0.4, n_2=0.15, r_1=0.2, r_2=0.040, f_s=0.001, f_t=0.1):
+        """
+        Initialize the snow mask task.
+        :param data_feature: EOPatch feature represented by a tuple in the form of `(FeatureType, 'feature_name')`
+            containing the bands B3, B4, B11 and B10
+
+            Example: (FeatureType.DATA, 'ALL-BANDS')
+
+        :type data_feature: tuple(FeatureType, str)
+        :param band_indices: A list containing the indices at which the required bands can be found the feature
+        :type band_indices: list(int)
+        :param cloud_mask_feature: EOPatch CLM feature represented by a tuple in the form of
+                                   `(FeatureType, 'feature_name')` containing the cloud mask
+        :type cloud_mask_feature: tuple(FeatureType, str)
+        :param dem_feature: EOPatch DEM feature represented by a tuple in the form of `(FeatureType, 'feature_name')`
+            containing the digital elevation model
+        :type dem_feature: tuple(FeatureType, str)
+        :param dilation_size: Size of the disk in pixels for performing dilation. Value 0 means do not perform
+                              this post-processing step.
+        :type dilation_size: int
+        :param r_f: Resize factor to produce the down-sampled red band
+        :type r_f: int
+        :param d_z: Size of elevation band in the digital elevation model used to define z_s
+        :type d_z: float
+        :param r_b: Minimum value of the red band reflectance to return a non-snow pixel to the cloud mask
+        :type r_b: float
+        :param r_d: Maximum value of the down-sampled red band reflectance to define a dark cloud pixel
+        :type r_d: float
+        :param n_1: Minimum value of the NDSI for the pass 1 snow test
+        :type n_1: float
+        :param n_2: Minimum value of the NDSI for the pass 2 snow test
+        :type n_2: float
+        :param r_1: Minimum value of the red band reflectance the pass 1 snow test
+        :type r_1: float
+        :param r_2: Minimum value of the red band reflectance the pass 2 snow test
+        :type r_2: float
+        :param f_s: Minimum snow fraction in the image to activate the pass 2 snow test
+        :type f_s: float
+        :param f_t: Minimum snow fraction in an elevation band to define z_s
+        :type f_t: float
+        """
+        self.r_f = r_f
```

Comment:
@AleksMat pylint complains that there are too many local variables, but all of these seem necessary, what would be the best thing to do? Increase the threshold for the check?

Reply:
@eerzin @mlubej maybe we could pass some of those variables as tuples. e.g `n_1` and `n_2` could be passed as `ndsi_threshold=(0.4, 0.15)`
This will also help by having input names easily untestable